rules/rules-reviewed/azure/springboot/spring-boot-to-azure-restricted-config.windup.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ruleset id="spring-boot-to-azure-restricted-config"
+         xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset identifies restricted configurations in Spring Boot applications for migration to Azure Spring Apps.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-xml,3.0.0.Final" />
+        </dependencies>
+        <sourceTechnology id="springboot"/>
+        <targetTechnology id="azure-spring-apps"/>
+    </metadata>
+    <rules>
+        <rule id="spring-boot-to-azure-restricted-config-01000">
+            <when>
+                <or>
+                    <filecontent filename="{*}.{extensions}" pattern="{pattern}"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Restricted configurations found" category-id="information" effort="0">
+                    <message>
+                        The application uses restricted configurations for Azure Spring Apps.
+                        These properties are automatically injected into your application environment by Azure Spring Apps to access Config Server and Service Discovery.
+                        Please remove them from your application, including configuration files, config server files, command line parameters, Java system attributes, and environment variables.
+
+                        If configured in **configuration files**: they will be ignored and overrided by Azure Spring Apps.
+                         
+                        If configured in **Config Server files**, **command line parameters**, **Java system attribute**, **environment variable**: they need to be removed or you might experience conflicts and unexpected behavior.
+                    </message>
+                    <link title="Remove restricted configurations" href="http://aka.ms/spring-cloud-to-asa?toc=%2Fazure%2Fspring-apps%2Ftoc.json&amp;bc=%2Fazure%2Fspring-apps%2Fbreadcrumb%2Ftoc.json&amp;pivots=sc-standard-tier#remove-restricted-configurations"/>
+                    <link title="Set up a Config server" href="https://learn.microsoft.com/azure/spring-apps/how-to-config-server"/>
+                    <link title="Enable Service Registration" href="https://learn.microsoft.com/azure/spring-apps/how-to-service-registration?pivots=programming-language-java"/>
+                </hint>
+            </perform>
+            <where param="extensions">
+                <matches pattern="(properties|yaml|yml)"/>
```

Comment:
You can leave YAML, but unfortunatelly this does not work yet: https://github.com/Azure/appcat-rulesets/issues/124

Reply:
> You can leave YAML, but unfortunatelly this does not work yet: #124

Awared. I also proposed a feature request 3 months ago but seems we still cannot parse yaml 😢[Add yamlfile condition for matching yamlpath](https://issues.redhat.com/browse/WINDUP-3876)